### PR TITLE
Update TXT record if A record changed

### PR DESCRIPTION
**Pull request by szok**

### CODE REVIEW

File: dnsaas/api/v2/views.py [hunk 1]
```diff
@@ -377,13 +377,25 @@ def _update_record(self, data):
             record.domain = hostname2domain(new['hostname'])
             record.content = new['address']
             record.save()
+            # If change hostname update name records txt.
+            log.debug('Update TXT records from: {} hostname to: {}'.format(
+                old['hostname'], new['hostname']
+            ))
+            Record.objects.filter(name=old['hostname']).update(
```
Comment (mkurek) [inline dnsaas/api/v2/views.py]: filter by type (TXT) too

File: dnsaas/api/v2/views.py [hunk 2]
```diff
@@ -377,13 +377,25 @@ def _update_record(self, data):
             record.domain = hostname2domain(new['hostname'])
             record.content = new['address']
             record.save()
+            # If change hostname update name records txt.
+            log.debug('Update TXT records from: {} hostname to: {}'.format(
```
Comment (mkurek) [inline dnsaas/api/v2/views.py]: log.info

File: dnsaas/api/v2/views.py [hunk 3]
```diff
@@ -377,13 +377,25 @@ def _update_record(self, data):
...
+                domain=record.domain
+            )
+
         return status.HTTP_200_OK, 'updated'
 
     def _delete_record(self, data):
         ip, hostname = data['address'], data['hostname']
         record = self._get_record(ip, hostname)
         if record:
+            log.debug('Delete record: {}'.format(record))
```
Comment (mkurek) [inline dnsaas/api/v2/views.py]: warning

File: dnsaas/api/v2/views.py [hunk 4]
```diff
@@ -377,13 +377,25 @@ def _update_record(self, data):
...
             record.delete()
+            log.debug('Delete TXT records for {} hostname'.format(hostname))
```
Comment (mkurek) [inline dnsaas/api/v2/views.py]: warning

File: dnsaas/api/v2/views.py [hunk 5]
```diff
@@ -377,13 +377,25 @@ def _update_record(self, data):
...
+            Record.objects.filter(name=hostname).delete()
```
Comment (mkurek) [inline dnsaas/api/v2/views.py]: filter by type (TXT) too

**Review by mkurek — CHANGES_REQUESTED** [review]

**Review by mkurek — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**